modules/react/common/stories/Responsive.stories.mdx

```diff
@@ -0,0 +1,314 @@
+`useResizeObserver | useResponsiveContainerStyles`
+
+`useResizeObserver` - This is a hook that will provide the width of the container. This can be used
+to measure container size as well.
+
+`useResponsiveContainerStyles` - This is a hook that will allow you to create container-based
+responsive styles with objects (as you can see in the example below). This hook accepts three
+arguments with the third being optional (Style Objects, Container Width, Theme). In each style
+object, there are five sizes that it will accept: `zero, small, medium, large and extra large`.
+These give sizes represent the breakpoints that are at the bottom of this page. The sizes will act
+like `min-width`. For example, if you want to apply styles from `medium` and up, then you would
+write those styles under `m`.
+
+In the example below, we are importing these utilities:
+`useResizeObserver | useResponsiveContainerStyles`
+
+See example below:
+
+```tsx
+import {Flex, Box} from '@workday/canvas-kit-react/layout';
+import {useResizeObserver, useResponsiveContainerStyles} from '@workday/canvas-kit-react/common';
+
+const ref = React.useRef(null);
+const [width, setWidth] = React.useState(0);
+
+useResizeObserver({
+  ref: ref,
+  onResize: data => {
+    setWidth(data.width || 0);
+  },
+});
+
+const containerResponsiveStyles = useResponsiveContainerStyles(
+  {
+    flex: {
+      flexDirection: 'column',
+      padding: 'm',
+      depth: 1,
+      borderRadius: 'l',
+      zero: {
+        backgroundColor: 'Red',
+      },
+      s: {
+        backgroundColor: 'Orange',
+      },
+      m: {
+        backgroundColor: 'Yellow',
+      },
+      l: {
+        backgroundColor: 'Green',
+      },
+      xl: {
+        backgroundColor: 'Blue',
+      },
+    },
+    box: {
+      padding: 's',
+    },
+  },
+  width
+);
+
+return (
+  <Box ref={ref}>
+    <Flex {...containerResponsiveStyles.flex}>
+      <Box {...containerResponsiveStyles.box}>Hello World</Box>
+    </Flex>
+  </Box>
+);
+```
+
+### Viewport Based Responsive Styling
+
+The viewport is the user's visible area of a web page and it varies with screen size.
+
+We can use media queries to run a series of tests that will measure the screen size or resolution
+and apply CSS styles specifically to an element.
+
+An example of CSS media queries is below:
+
+```css
+@media screen and (min-width: 800px) {
+  .container {
+    margin: 1em 2em;
+  }
+}
+```
+
+In the example below, viewport-based styles are being applied by the size of the user's viewport.
+This is by using the breakpoint utilities (`up(), down(), between() and only()`). At the top of the
+code you will see `up() and down()` being used inside of the `StyledHeading` component. These
+utilities will be explained more in depth below the example. The size of the browser will have to be
+adjusted to see the changes take place on the UI Layout.
+
+<ExampleCodeBlock code={ResponsiveViewport} />
+
+### Applying Viewport-Based Responsive Styling
```

Comment:
Let's just call this section Breakpoint Utilities.
```suggestion
### Breakpoint Utilities
```

Reply:
Agreed